cmd/block-generator/runner/run.go

```diff
@@ -0,0 +1,260 @@
+package runner
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	// Load the postgres sql.DB implementation
+	_ "github.com/lib/pq"
+
+	"github.com/algorand/indexer/cmd/block-generator/generator"
+	"github.com/algorand/indexer/util"
+)
+
+// Args are all the things needed to run a performance test.
+type Args struct {
+	// Path is a directory when passed to RunBatch, otherwise a file path.
+	Path                     string
+	IndexerBinary            string
+	IndexerPort              uint64
+	PostgresConnectionString string
+	RunDuration              time.Duration
+	ReportDirectory          string
+
+	indexerPort uint64
+}
+
+// Run is a public helper run the tests.
+// The test will run against the generator configuration file specified by 'args.Path'.
+// If 'args.Path' is a directory it should contain generator configuration files, a test will run using each file.
+func Run(args Args) error {
+	if _, err := os.Stat(args.ReportDirectory); !os.IsNotExist(err) {
```

Comment:
Turns out, I'm not sure the point of `os.IsExist`. This interface is not very good for checking that a directory exists because the check would need to be `err == nil || os.IsExist(err)`!

I'll grant that the double negative is ugly, but the alternative is too!